paperscraper/lib.py

```diff
@@ -387,7 +394,8 @@ async def doi_to_bibtex(doi: str, session: ClientSession) -> str:
     # get DOI via crossref
     url = f"https://api.crossref.org/works/{doi}/transform/application/x-bibtex"
     async with session.get(url) as r:
-        r.raise_for_status()
+        if not r.ok:
+            raise DOINotFoundError(f"Could not resolve DOI {doi}")
```

Comment:
Two comments here are:
- With a custom exception here (compared with `raise_for_status`), we lose status code here. Let's keep that in the error message
- If `doi` is an empty string, the message is harder to read, so try to put a period after the format specifier

```python
if not r.ok:
    raise DOINotFoundError(f"Per HTTP status code {r.status_code}, could not resolve DOI {doi}.")
```

Reply:
yup - fixed